src/snowflake/snowpark/modin/plugin/compiler/snowflake_query_compiler.py

```diff
@@ -2373,7 +2373,11 @@ def _reindex_axis_0(
         """
         self._raise_not_implemented_error_for_timedelta()
 
-        new_index_qc = pd.Series(labels)._query_compiler
+        if isinstance(labels, (pd.Index)):
```

Comment:
No what I mean was that
If it is native index, we can call pd.Index(native_idx) to turn it lazy
and then we can call to_series to turn it to series and also the index name should be kept.

Reply:
fixed!